From d8d0cee6cffede9cf853c6835ae16e91e25dbecf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?K=C3=A9vin=20L=27h=C3=B4pital?=
 <kevin.lhopital@savoirfairelinux.com>
Date: Thu, 31 Mar 2022 13:01:42 +0200
Subject: [PATCH] disable libx11 dependency

This dependency is irrelevant on i.MX6 platforms because we don't use
X11.
Moreover compiling with libX11 requires to enable the disto feature X11
which will change the compilation of Qt5 and disable the EGLFS_VIV
plugin.
---
 CMakeLists.txt    |  6 +-----
 src/avadapter.cpp | 26 --------------------------
 2 files changed, 1 insertion(+), 31 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index aec39c0..10f2498 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -279,13 +279,11 @@ if(MSVC)
     )
 elseif (NOT APPLE)
     list(APPEND COMMON_SOURCES
-                ${SRC_DIR}/xrectsel.c
                 ${SRC_DIR}/dbuserrorhandler.cpp
                 ${SRC_DIR}/connectivitymonitor.cpp
                 ${SRC_DIR}/updatemanager.cpp)
     list(APPEND COMMON_HEADERS
-                ${SRC_DIR}/dbuserrorhandler.h
-                ${SRC_DIR}/xrectsel.h)
+                ${SRC_DIR}/dbuserrorhandler.h)
     list(APPEND QT_LIBS Qt::DBus)
     list(APPEND QT_MODULES DBus)
 
@@ -360,7 +358,6 @@ elseif (NOT APPLE)
 
     find_library(ringclient ringclient ${LRCLIBDIR} NO_DEFAULT_PATH)
     find_library(qrencode qrencode)
-    find_library(X11 X11)
 else() # APPLE
     list(APPEND COMMON_SOURCES
                 ${SRC_DIR}/os/macos/updatemanager.mm
@@ -528,7 +525,6 @@ elseif (NOT APPLE)
                           ${QT_LIBS}
                           ${LRC_LIB_NAME}
                           ${qrencode}
-                          ${X11}
                           ${LIBNM_LIBRARIES}
                           ${LIBNOTIFY_LIBRARIES}
                           ${LIBGDKPIXBUF_LIBRARIES}
diff --git a/src/avadapter.cpp b/src/avadapter.cpp
index c46836e..ea8768b 100644
--- a/src/avadapter.cpp
+++ b/src/avadapter.cpp
@@ -24,10 +24,6 @@
 #include "api/newcodecmodel.h"
 #include "api/newdevicemodel.h"
 
-#ifdef Q_OS_LINUX
-#include "xrectsel.h"
-#endif
-
 #include <QtConcurrent/QtConcurrent>
 #include <QApplication>
 #include <QPainter>
@@ -192,27 +188,6 @@ AvAdapter::shareFile(const QString& filePath)
 void
 AvAdapter::shareScreenArea(unsigned x, unsigned y, unsigned width, unsigned height)
 {
-#ifdef Q_OS_LINUX
-    // xrectsel will freeze all displays too fast so that the call
-    // context menu will not be closed even closed signal is emitted
-    // use timer to wait until popup is closed
-    QTimer::singleShot(100, [=]() mutable {
-        x = y = width = height = 0;
-        xrectsel(&x, &y, &width, &height);
-        auto resource = lrcInstance_->getCurrentCallModel()->getDisplay(getScreenNumber(),
-                                                                        x,
-                                                                        y,
-                                                                        width < 128 ? 128 : width,
-                                                                        height < 128 ? 128 : height);
-        auto callId = lrcInstance_->getCurrentCallId();
-        lrcInstance_->getCurrentCallModel()
-            ->requestMediaChange(callId,
-                                 "video_0",
-                                 resource,
-                                 lrc::api::NewCallModel::MediaRequestType::SCREENSHARING,
-                                 false);
-    });
-#else
     auto resource = lrcInstance_->getCurrentCallModel()->getDisplay(getScreenNumber(),
                                                                     x,
                                                                     y,
@@ -225,7 +200,6 @@ AvAdapter::shareScreenArea(unsigned x, unsigned y, unsigned width, unsigned heig
                              resource,
                              lrc::api::NewCallModel::MediaRequestType::SCREENSHARING,
                              false);
-#endif
 }
 
 void
-- 
2.25.1

